test/app-luatest/preload_test.lua

```diff
@@ -0,0 +1,354 @@
+local fun = require('fun')
+local fio = require('fio')
+local log = require('log')
+local json = require('json')
+local popen = require('popen')
+
+local t = require('luatest')
+local g = t.group()
+
+-- Find an item in an array table and return its index.
+--
+-- array_find('b', {'a', 'b', 'c'}) -> 2
+-- array_find('x', {'a', 'b', 'c'}) -> nil
+local function array_find(needle, array)
+    for i, item in ipairs(array) do
+        if item == needle then
+            return i
+        end
+    end
+    return nil
+end
+
+-- All Lua files produced by the test write a JSON object to
+-- stdout. This object contains everything we want to know
+-- about the context of the script execution.
+--
+-- First of all, we should ensure that given script was executed.
+-- So we output a script file name.
+--
+-- Next, we should define what is passed to the script as `...`
+-- (dots).
+--
+-- We should also inspect `arg` content to freeze certain
+-- behavior.
+--
+-- Note: In order to serialize `arg` into JSON it is split into
+-- `arg[-1]`, `arg[0]` and `arg[]`. The latter means an array of
+-- all items from 1 to the end.
+local SCRIPT_TEMPLATE = [[
+print(require('json').encode({
+    ['script'] = '<script>',
+    ['...'] = {...},
+    ['arg[-1]'] = arg[-1],
+    ['arg[0]'] = arg[0],
+    ['arg[]'] = setmetatable(arg, {__serialize = 'seq'}),
+}))
+]]
+
+local tempdirs = {}
+
+-- Remove all temporary directories created by the test
+-- unless KEEP_DATA environment variable is set.
+g.after_all = function()
+    local dirs = table.copy(tempdirs)
+    tempdirs = {}
+
+    local keep_data = (os.getenv('KEEP_DATA') or '') ~= ''
```

Comment:
Minor: You can just write `local keep_data = not not os.getenv('KEEP_DATA')`.

Feel free to ignore.